### PR TITLE
mobile: switching idle timeout test to full network stack

### PR DESCRIPTION
**Pull request by alyssawilk**
Changing a test to use a fake upstream and time out sending the request.

Risk Level: n/a (test only) 
Testing: yes
Docs Changes: no

### CODE REVIEW

File: mobile/test/swift/integration/IdleTimeoutTest.swift [hunk 1]
```diff
@@ -157,27 +85,30 @@ static_resources:
...
+    let requestHeaders = RequestHeadersBuilder(
+      method: .get, scheme: "http", authority: "localhost:" + port, path: "/"
+    )
+    .build()
 
     client
       .newStreamPrototype()
       .setOnError { error, _ in
-        XCTAssertEqual(error.errorCode, 0)
+        XCTAssertEqual(error.errorCode, 4)
```
Comment (alyssawilk) [inline mobile/test/swift/integration/IdleTimeoutTest.swift:97]: Apparently I'm testing a different timeout for 4 is ENVOY_REQUEST_TIMEOUT which is an idle timeout on the request path, triggerd by sending no end stream.

**Comment by alyssawilk** [comment]
Ok one last listener change.  The prior push made sure this wouldn't break CI with a merge conflict with https://github.com/envoyproxy/envoy/pull/30698 so it's good to automerge.

**Review by abeyad — APPROVED** [review]
awesome!

**Comment by abeyad** [comment]
The CI failures look legit:
```
Code coverage for source/common/api is lower than limit of 85.8 (84.5)
Code coverage for source/common/api/posix is lower than limit of 85.2 (83.8)
```

But it's unclear how your PR would've caused it.

**Comment by RyanTheOptimist** [comment]
> The CI failures look legit:
> 
> ```
> Code coverage for source/common/api is lower than limit of 85.8 (84.5)
> Code coverage for source/common/api/posix is lower than limit of 85.2 (83.8)
> ```
> 
> But it's unclear how your PR would've caused it.

I think coverage is currently flaky. https://github.com/envoyproxy/envoy/pull/30701

**Pull request merged**